ReactNative/ReactTwitter/core/service/deep-linking-facade.ios.js

```diff
@@ -0,0 +1,30 @@
+import { Linking } from 'react-native'
+
+class DeepLinkingFacade {
+
+  static newInstance () {
+    return new DeepLinkingFacade(Linking)
+  }
+
+  constructor (linking) {
+    this.linking = linking
+  }
+
+  listenForDeepLinking () {
+    this.linking.removeEventListener('url', this._handleOpenURL)
+    this.linking.addEventListener('url', this._handleOpenURL)
+    return new Promise(function (resolve, reject) {
```

Comment:
We wanted to be sure we don't register several times to the event listener. We end up with something that is not really clear, probably we should use a instance variable to know if we are already registered or not